### PR TITLE
New SKU support for MSN3420

### PR DESCRIPTION
**Pull request by shlomibitton**
**- What I did**
Added and modified files to support new SKU MSN3420.
Added the support for this SKU in platform API.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers_defaults_t0.j2 [hunk 1]
```diff
@@ -0,0 +1,104 @@
+{% set default_cable = '5m' %}
```
Comment (keboliu) [inline device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers_defaults_t0.j2:1]: just set a symbol link to the file of 3700, they are the same

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers_defaults_t1.j2 [hunk 2]
```diff
@@ -0,0 +1,104 @@
+{% set default_cable = '5m' %}
```
Comment (keboliu) [inline device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers_defaults_t1.j2:1]: just set a symbol link to the file of 3700, they are the same

File: device/mellanox/x86_64-mlnx_msn3420-r0/platform_reboot [hunk 3]
```diff
@@ -0,0 +1 @@
+../x86_64-mlnx_msn3700-r0/platform_reboot
```
Comment (keboliu) [inline device/mellanox/x86_64-mlnx_msn3420-r0/platform_reboot:1]: it should link to  ../../x86_64-mlnx_msn2700-r0/platform_reboot    and same comments to below symbol link, please go through.

File: device/mellanox/x86_64-mlnx_msn3420-r0/plugins/sfputil.py [hunk 4]
```diff
@@ -0,0 +1 @@
+../../x86_64-mlnx_msn3700-r0/plugins/sfputil.py
```
Comment (keboliu) [inline device/mellanox/x86_64-mlnx_msn3420-r0/plugins/sfputil.py:1]: as above comments, it should be link to 2700 plugin and, should change the source file to add this new SKU   https://github.com/Azure/sonic-buildimage/blob/master/device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py#L44

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# name          lanes               alias
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini:1]: @shlomibitton please add `index` column:
https://github.com/Azure/sonic-buildimage/blob/master/device/mellanox/x86_64-mlnx_msn2010-r0/ACS-MSN2010/port_config.ini

File: device/mellanox/x86_64-mlnx_msn3420-r0/platform_components.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+    "chassis": {
+        "x86_64-mlnx_msn3420-r0": {
+            "component": {
+                "BIOS": { },
+                "CPLD": { }
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn3420-r0/platform_components.json:6]: @shlomibitton please add the relevant number of platform CPLDs (e.g., CPLD1, CPLD2, CPLD3, etc)

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers_defaults_t0.j2 (hunk 1 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers_defaults_t0.j2]: changed symbol link

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers_defaults_t1.j2 (hunk 2 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/buffers_defaults_t1.j2]: changed symbol link

File: device/mellanox/x86_64-mlnx_msn3420-r0/platform_reboot (hunk 3 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/platform_reboot]: changed symbol link

File: device/mellanox/x86_64-mlnx_msn3420-r0/plugins/sfputil.py (hunk 4 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/plugins/sfputil.py]: changed symbol link

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini (hunk 5 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini]: Updated column 'index'

File: device/mellanox/x86_64-mlnx_msn3420-r0/platform_components.json (hunk 6 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/platform_components.json:6]: Added 3 CPLD's

**Comment by shlomibitton** [comment]
@nazariig @keboliu 
please review

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+Ethernet200     200,201,202,203     etp51               50
+Ethernet204     204,205,206,207     etp52               51
+Ethernet208     208,209,210,211     etp53               52
+Ethernet212     212,213,214,215     etp54               53
+Ethernet216     216,217,218,219     etp55               54
+Ethernet220     220,221,222,223     etp56               55
+Ethernet224     224,225,226,227     etp57               56
+Ethernet228     228,229,230,231     etp58               57
+Ethernet232     232,233,234,235     etp59               58
+Ethernet236     236,237,238,239     etp60               59
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini:61]: @shlomibitton please have the same indentation: 4 spaces

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/sai_3420.xml [hunk 8]
```diff
@@ -0,0 +1,439 @@
+<?xml version="1.0"?>
+<root>
+    <platform_info type="3420">
+
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/sai_3420.xml:4]: @shlomibitton what about ISSU?

File: device/mellanox/x86_64-mlnx_msn3420-r0/platform_components.json (hunk 6 above)
Comment (nazariig) [reply device/mellanox/x86_64-mlnx_msn3420-r0/platform_components.json:6]: @shlomibitton please double check. This one is still not fixed.

File: device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini (hunk 7 above)
Comment (shlomibitton) [reply device/mellanox/x86_64-mlnx_msn3420-r0/ACS-MSN3420/port_config.ini]: Fixed indentation.

**Pull request merged**